accepted/2021/platform-dependencies/platform-dependencies.md

```diff
@@ -0,0 +1,433 @@
+
+Luckily, the burden of maintenance is expected to be low since dependencies don't often change. However, there are a few scenarios that do cause change which are described further below.
+
+### Maintenance Scenarios
+
+In all of the scenarios below, the person introducing or discovering the dependency change should file an issue in the [dotnet/core](https://github.com/dotnet/core) repo describing the change. Such issues should be labeled as `area-dependency`. This is necessary in order to notify stakeholder teams (e.g. installers, container images, documentation) of such changes.
+
+#### New Dependency
+
+New platform dependencies can be introduced by making a code change that has the dependency or including a new component into .NET (e.g. shared framework, NuGet package). When such a dependency is added, the contributor should also update the appropriate platform dependency model file with the necessary information that describes the dependency across all supported platforms.
+
+##### Change Detection
+
+In order to avoid the reliance upon contributors to recognize when they've changed a dependency, a more automated solution would be preferable. This can be done by defining a GitHub bot that checks for files in PRs containing `NativeMethods` or `Interop` in their name. If such a file is detected, a label is added to the PR alerting the submitter that they should evaluate their changes for changes to the dependencies. While not a fool-proof solution, it should provide coverage for the vast majority of dependencies. Work is still required by the submitter to make the appropriate changes to the platform dependency model but the GitHub bot helps alert them when there is potentially action that is needed.
+
+#### Dependency Name Change
+
+The platform dependency model describes dependencies by name and those names are subject to change from upstream sources.
+
+> A classic example of this is the [ICU library](http://site.icu-project.org/). In many Linux distros, the ICU library's package contains the version number in the name (e.g. libicu57). When a Linux distro updates to use a new version of the ICU library, they typically remove the previous version from the package repository. Thus, any dependency described for the ICU library needs to account for the name change.
+
+We will rely on manual discovery to determine when the name of a package has changed since the typical scenario in which a name change is detected happens when a build breaks. When a dependency name change is discovered, the person discovering the change should file an issue in dotnet/core as described [above](#maintenance-scenarios).
+
+#### New Platform Support
+
+When support for a new platform is added to the product, the platform dependency model of future releases needs to be updated include this platform and all its supported versions.
```

Comment:
Is there any impact on unsupported platforms? For example, we are trying to get source-build to work with Arch Linux. If those folks get .NET building on Arch, will they be affected if we start tracking dependencies here? Will anything suddenly start breaking for them? And will it be okay if we don't want to track Arch Linux here if we dont want to "officially support" it?

Reply:
No, nothing would break.  This is purely a representation of what dependencies exist; it doesn't dictate what those dependencies must be.  The flow of information would always be that the maintainers of the .NET source define what dependencies they want to have, and then the dependency model gets updated to reflect that.  There would be no use of the dependency model at runtime or build time (i.e. it has no impact on the running of a .NET application or of the building of .NET source). Its purpose is to add rigor and correctness to the maintenance of other assets that describe what the dependencies are (such as documentation).